### DISCUSSION TITLE
Can’t fetch a branch I pushed on another computer

### DISCUSSION THREAD

**Discussion by wjk** [community/community#91661] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have a repository on GitHub, [wjk/ReactOS](https://github.com/wjk/ReactOS). I have performed a shallow clone of this repo on my two dev boxes. (I must do a shallow clone as the repository is far too large to clone any other way.) On computer A I have created and pushed a branch, named `control-panel-canonical-names`. This branch is fully visible on github.com. However, I cannot check this branch out on computer B. When I run `git checkout` there, I get an error stating `pathspec: '<branch name>' did not match any file(s)
known to git`. `git branch <name> origin/<name>` doesn't work either; it complains that the second parameter is not a valid object name. `git branch -a` lists many branches from another remote (my upstream), but not the branch I want or any other branches that exist in my fork, all of which exhibit this same behavior. I have already tried `git fetch --all`; it hasn't helped. What can I do about this?

**Comment by NipulM** [comment]
On computer B, try fetching the branch specifically from the remote repository:
`git fetch origin control-panel-canonical-names
`

After fetching, try checking out the branch again:
`git checkout control-panel-canonical-names
`


if that doesn't work, maybe try to update the shallow clone with a deeper history?
`git fetch --unshallow
`

And then do a:
`git checkout control-panel-canonical-names
`

Hopefully, this resolves the issue for you! 🤞🏼

**Reply by wjk** [reply]
`git fetch origin ...` worked, but I still cannot checkout the branch (same symptoms). I already tried `git fetch --unshallow` and it clearly didn't work, because when I tried running that the computer said the repo isn't shallow.

**Reply by NipulM** [reply]
I'm assuming you've already created the branch locally. If its still not tracking the remote branch, you can set the upstream explicitly using:
`git branch --set-upstream-to=origin/control-panel-canonical-names control-panel-canonical-names
`
After this, you should be able to simply use `git checkout control-panel-canonical-names`.

If you haven't created a local branch yet, you can try:
`git checkout -b control-panel-canonical-names origin/control-panel-canonical-names
`
This creates a new _local branch_  and sets it to track the remote branch. 

Hope this helps!

**Reply by wjk** [reply]
Still doesn't work.

```shell
$ git branch -u origin/control-panel-canonical-names control-panel-canonical-names
fatal: branch 'control-panel-canonical-names' does not exist

$ git checkout -b control-panel-canonical-names origin/control-panel-canonical-names
fatal: 'origin/control-panel-canonical-names' is not a commit and a branch 'control-panel-canonical-names' cannot be created from it
```

**Reply by NipulM** [reply]
It seems like the remote branch is not being recognised as a valid branch.

try doing:
```
git checkout --orphan control-panel-canonical-names      `this gonna create a local branch without setting up tracking`
git reset --hard      `Discard any changes (if any)`  
git fetch origin control-panel-canonical-names

git reset --hard origin/control-panel-canonical-names      `Reset the branch to the fetched commit`
git branch -u origin/control-panel-canonical-names control-panel-canonical-names      `Now set up tracking for the branch`
git checkout control-panel-canonical-names      
```

**Reply by wjk** [reply]
Still doesn't work.

```shell
$ git branch -u origin/control-panel-canonical-names control-panel-canonical-names
fatal: branch 'control-panel-canonical-names' does not exist
```

**Reply by NipulM** [reply]
This is really interesting. 

First, create the local branch from the remote branch:
`git checkout -b control-panel-canonical-names origin/control-panel-canonical-names
`

After that explicitly set up the tracking:
`git branch --set-upstream-to=origin/control-panel-canonical-names control-panel-canonical-names
`

This should track the remote branch. If you encounter any issues or if the problem persists please let me know!

**Reply by wjk** [reply]
I already tried that. It isn’t working. As I understand it `-u` and `--set-upstream-to=` are identical.

**Reply by wjk** [reply, marked as answer]
I have since fixed this by recloning with `--filter=blob:none`, which downloaded very quickly.
